### ISSUE TITLE
Why does pyaff4 needs aff4-snappy instead of python-snappy?

### ISSUE THREAD

**Issue by joachimmetz** [aff4/pyaff4#9]
pyaff4 depends on aff4-snappy
* https://pypi.org/project/aff4-snappy/
* https://github.com/rekall-innovations/aff4-snappy

Which is forked from python-snappy:
* https://github.com/andrix/python-snappy
* https://pypi.org/project/python-snappy/

@blschatz @scudette I'm looking into building pyaff4 for testing purposes, some questions:
* What makes aff4-snappy different from python-snappy?
* Could the changes in aff4-snappy be merged into python-snappy?
* Could pyaff4 be changed to use python-snappy instead of aff4-snappy?

**Comment by scudette** [comment]
python snappy is not a hermetic distribution - it requires libsnappy to be installed on the system. It then builds the python binding with a runtime requirement for libsnappy. This is not an acceptable solution.

The main change is this commit https://github.com/rekall-innovations/aff4-snappy/commit/45397f786b7caea874db9f79faf7b6b4bf10d0a9

Modified to build snappy inline so we do not need to link against libsnappy.

For something as trivial as snappy it does not make sense to include a shared object - just build it statically into the binding.

**Comment by joachimmetz** [comment]
thx for the context

Let me know if you're OK with the following changes:

1. Move aff4-snappy under https://github.com/aff4?
2. Change aff4-snappy to have the libsnappy as a gitsubmodule (similar to pytsk)? So it easier to build against newer version of libsnappy
3. Update aff4-snappy to latest version of python-snappy. Push all non build related changes upstream
4. Include libsnappy and python-snappy licenses in sdist

**Comment by joachimmetz** [comment]
@scudette @blschatz friendly ping any feedback/input?

**Comment by scudette** [comment]
Thanks for the ping I totally missed it :-).

Python-snappy builds a shared object which depends on the system
libsnappy.so or libsnappy.dll. this makes it not hermetic so we can not use
it to distribute on pipy.

The aff4-snappy fork builds the bindings and the snappy code into a self
contained module.

This might have changed recently I have not looked at the package for a
long time. But you can verify it using ldd on the produced .so file, or try
to build on Windows which will fail unless you have snappy installed.

Thanks
Mike

**Comment by joachimmetz** [comment]
My point was more to change pyaff4 to include snappy as is done for pytsk (as a git submodule) and create an hermetic build?

**Comment by scudette** [comment]
Ah sorry I don't read the entire thread (just send an email).  Sure what you suggest sounds great. Will accept any pr to do this :-)

**Comment by joachimmetz** [comment]
Happy to help but could you maybe follow up on the steps proposed in https://github.com/aff4/pyaff4/issues/9#issuecomment-485725653
